src/lua/httpc.lua

```diff
@@ -420,7 +458,12 @@ curl_mt = {
                 resp.headers = process_headers(resp.headers)
             end
 
-            return resp
+            return setmetatable(resp, {
+                __index = {
+                    decode = decode_body,
+                    decoders = self.decoders,
```

Comment:
TBH, I don't see a reason why the decoders table is placed here. It is accessible from the HTTP client object and it seems enough.